colossalai/booster/booster.py

```diff
@@ -107,6 +108,15 @@ def __init__(
         else:
             self.checkpoint_io = GeneralCheckpointIO()
 
+    def setup_distributed_optim(self, optim):
+        """Helper to setup optimizers that require global states"""
+        shard_to_param = (
+            optim._param_store.master_to_working_param if isinstance(optim, LowLevelZeroOptimizer) else None
+        )
+        tp_group = self.plugin.__dict__.get("tp_group", None)
+        dp_group = self.plugin.__dict__.get("dp_group", None)
+        optim.optim.setup_distributed(tp_group, dp_group, shard_to_param)
+
```

Comment:
Add `support_distributed_optim()` method for plugins, like
https://github.com/hpcaitech/ColossalAI/blob/079bf3cb26a502fc647b1aad15fd14d6266be66c/colossalai/booster/plugin/plugin_base.py#L33

I think `optim.setup_distributed()` should be called in `plugin.configure()`

You should do sanity check here. E.g. raise an error if plugin is not used.